pkg/controller/dns/external_dns.go

```diff
@@ -88,27 +92,50 @@ func instances(conf *config.Config) []instance {
 			resources: privateResources,
 			action:    privateAction,
 		},
-	}
+	}, nil
 }
 
-func publicConfig(conf *config.Config) *manifests.ExternalDnsConfig {
-	return &manifests.ExternalDnsConfig{
-		TenantId:           conf.TenantID,
-		Subscription:       conf.PublicZoneConfig.Subscription,
-		ResourceGroup:      conf.PublicZoneConfig.ResourceGroup,
-		Provider:           manifests.PublicProvider,
-		DnsZoneResourceIDs: util.Keys(conf.PublicZoneConfig.ZoneIds),
+func publicConfigForIngress(conf *config.Config) *manifests.ExternalDnsConfig {
+	publicconfig, err := manifests.NewExternalDNSConfig(
+		conf,
+		conf.TenantID,
+		conf.PublicZoneConfig.Subscription,
+		conf.PublicZoneConfig.ResourceGroup,
+		conf.MSIClientID,
+		"",
+		conf.NS,
+		"",
+		manifests.IdentityTypeMSI,
+		[]manifests.ResourceType{manifests.ResourceTypeIngress},
+		manifests.PublicProvider,
+		util.Keys(conf.PublicZoneConfig.ZoneIds))
+
+	if err != nil {
+		// do something
```

Comment:
do something

Reply:
did something :D sorry about this not sure how it got rolled back lol